File: cbp/potega.cpp
/*
 * potega.cpp
 * a0=1
 * a1 = a
 * 
 */


#include <iostream>

using namespace std;

float potega_it(float x, int n) {
    float wynik = 1;
    for (int i=1 ;i<=n ;i++ ) {
        //testuję ilośc powtórzeń pętli
        wynik = wynik * x;
        
    }
    return wynik;
    }
    
        
int main(int argc, char **argv)
{ 
    int n;
    float x;
    cout << "Podaj podstawę potęgi: " << endl;
    cin >> x;
    cout << "Podaj wykładnik potęgi: " << endl;
    cin >> n;
 	//pobierz od uzytkownika podstawę i wykładnik
	cout << "Potęga: " << potega_it(x, n) << endl;
    return 0;
    
}

File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py

def potega_it(x, n):

    i = 1
    wynik = 1
    while i <= n:
        wynik = wynik * x
        i = i + 1

    return wynik

def potega_rek(a, n):
    if x == 0:
        return 0
    elif n == 0:
        return 1
        return potega_rek(x, n-1)* x




def main(args):
    x = float(input("Podaj podstawe potęgi: "))
    n = int(input("Podaj wykładnik potęgi: "))
    print('Wynik: ', potega_it(x, n))
    return 0

if __name__ == '__main__':
    import sys
sys.exit(main(sys.argv))
